tools/stonith_admin.c

```diff
@@ -639,30 +613,32 @@ main(int argc, char **argv)
     switch (action) {
         case 'I':
             rc = st->cmds->list_agents(st, st_opt_sync_call, NULL, &devices, timeout);
```

Comment:
I just noticed we do need to handle one more case here and below, when rc < 0. devices stays NULL in that case, so previously it would just not output anything (aside from stderr messages that the stonith API commands might emit) and exit based on rc. However, we now unconditionally set rc to 0, so any error information is lost. We can just check rc < 0 here and break (although it might be nice to print something to stderr with pcmk_strerror()).

Reply:
Just breaking will cause crm_info to get called just before the done label.  Is that enough, or should I also print to stderr there?

Reply:
crm_info() is only seen if the user gave multiple -V options; stderr is better for anything the user really should see.